backend/api/Services/LocalizationService.cs

```diff
@@ -0,0 +1,247 @@
+
+            timer.Start();
+            while (timer.Elapsed.TotalSeconds < Timeout)
+            {
+                if (ongoingMissionRun is null) { continue; }
+                if (ongoingMissionRun.Status == MissionStatus.Successful) { return; }
+
+                ongoingMissionRun = await missionRunService.ReadById(ongoingMissionRunId);
+            }
+
+            const string Message = "Timed out while waiting for the localization mission to get an updated status";
+            logger.LogError("{Message}", Message);
+            throw new TimeoutException(Message);
+        }
+
+        private async Task<string> StartLocalizationMissionInArea(string robotId, string areaId)
+        {
+            var robot = await robotService.ReadById(robotId);
+            if (robot is null)
+            {
+                string errorMessage = $"The robot with ID {robotId} was not found";
+                logger.LogError("{Message}", errorMessage);
+                throw new RobotNotFoundException(errorMessage);
+            }
+
+            var area = await areaService.ReadById(areaId);
+            if (area is null)
+            {
+                string errorMessage = $"The area with ID {areaId} was not found";
+                logger.LogError("{Message}", errorMessage);
+                throw new AreaNotFoundException(errorMessage);
+            }
+
+            if (area.Deck?.DefaultLocalizationPose?.Pose is null)
+            {
+                const string ErrorMessage = "The mission area is not associated with any deck or that deck does not have a localization pose";
+                logger.LogError("{Message}", ErrorMessage);
+                throw new DeckNotFoundException(ErrorMessage);
+            }
+            if (robot.Status is not RobotStatus.Available)
+            {
+                string errorMessage = $"Robot '{robot.Id}' is not available as the status is {robot.Status}";
+                logger.LogWarning("{Message}", errorMessage);
+                throw new RobotNotAvailableException(errorMessage);
+            }
+
+            var localizationMissionRun = new MissionRun
+            {
+                Name = "Localization mission",
+                Robot = robot,
+                InstallationCode = area.Installation.InstallationCode,
+                Area = area,
+                Status = MissionStatus.Pending,
+                DesiredStartTime = DateTime.UtcNow,
+                Tasks = new List<MissionTask>
+                {
+                    new(area.Deck.DefaultLocalizationPose.Pose, "localization")
+                },
+                Map = new MapMetadata()
+            };
+            logger.LogWarning("Starting localization mission");
+            await missionRunService.Create(localizationMissionRun, triggerCreatedMissionRunEvent: false);
+            await robotService.UpdateCurrentArea(robot.Id, localizationMissionRun.Area);
```

Comment:
Only update current area once the localization mission has been successfully run. To avoid scheduling multiple localization missions we could check the mission queue for a pending or ongoing localization mission

Reply:
We should not be scheduling multiple localization missions now? I think this is a good improvement but would suggest adding it as an improvement in the next PR.